src/models/step_arguments/data_table.js

```diff
@@ -1,5 +1,31 @@
 import _ from 'lodash'
 
+const truths = ['true', 'TRUE', 'True', 'yes', 'YES', 'Yes', 'y', 'Y', '1']
+const trueFalse = (v) => !!~truths.indexOf(v)
+const asIs = (v) => v
+const list = (v) => v.split(',')
+const date = (v) => new Date(v)
+
+const types = {
+  'string':  asIs,
+  'str':     asIs,
+  '':        asIs,
+  'int':     parseInt,
+  'integer': parseInt,
+  'double':  parseInt,
+  'float':   parseFloat,
+  'number':  parseFloat,
+  'bool':    trueFalse,
+  'boolean': trueFalse,
+  'y/n':     trueFalse,
+  'bit':     trueFalse,
+  date,
+  'datetime':date,
+  'array':   list,
+  list,
+  'json':    JSON.parse
```

Comment:
Thoughts on the following: we could define the types as I suggested in order to dry up the configuration and then programmatically create the mapping once in order to keep it performant. I like the dried up version as its easier to see what is equivalent.